phost/src/types.rs

```diff
@@ -69,31 +70,83 @@ impl Resp for GetInfoReq {
     type Resp = GetInfoResp;
 }
 
+#[derive(Serialize, Deserialize, Debug, Clone)]
+pub enum Payload {
+    Plain(String),
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+pub struct Query {
+    pub contract_id: u32,
+    pub nonce: u32,
+    pub request: ReqData,
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+pub enum ReqData {
+    TransferToChain {sequence: u32}
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+pub struct QueryReq {
+    pub query_payload: String,
+}
+#[derive(Serialize, Deserialize, Debug)]
+#[serde(rename_all = "PascalCase")]
+pub struct QueryResp {
+    pub plain: String,
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+#[serde(rename_all = "PascalCase")]
+pub struct TransferToChain {
+    pub transfer_to_chain: TransferQueue,
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+pub struct TransferQueue {
+    pub transfer_queue: Vec<Vec<u8>>,
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+#[derive(Encode, Decode)]
+pub struct TransferData {
```

Comment:
We'd better to define it only once somewhere and reference it in other places. You can feel free to leave it to future refactor, but let's add a `// TODO` here (also, don't forget phost)

Reply:
leave it to future refactor